mos_tests/rabbitmq_oslo/conftest.py

```diff
@@ -38,3 +44,43 @@ def fixt_kill_rpc_server_client(env):
         with node.ssh() as remote:
             cmd = 'pkill -f oslo_msg_check_'
             remote.execute(cmd)
+
+
+@pytest.fixture
+def controller(env):
+    return random.choice(env.get_nodes_by_role('controller'))
+
+
+@pytest.yield_fixture
+def patch_iptables(controller, request):
+    """Apply IPTables rules"""
+
+    def add_ports(tmpl):
+        ports = [4369, 5672, 5673, 25672]
+        return ' '.join(map(lambda portnum: tmpl.format(portnum=portnum),
+                            ports))
+
+    if request.param == 'drop':
+        tbl_modif = add_ports(
+            'iptables -I INPUT -p tcp -m tcp --dport {portnum} -j DROP ;')
+        tbl_modif_del = add_ports(
+            'iptables -D INPUT -p tcp -m tcp --dport {portnum} -j DROP ;')
+    elif request.param == 'reject':
+        tbl_modif = add_ports(
+            'iptables -I INPUT -p tcp -m tcp --dport {portnum} -j REJECT'
+            ' --reject-with icmp-host-prohibited ;')
+        tbl_modif_del = add_ports(
+            'iptables -D INPUT -p tcp -m tcp --dport {portnum} -j REJECT'
+            ' --reject-with icmp-host-prohibited ;')
+    else:
+        raise ValueError("Does'n know such param [{0}]!".format(request.param))
+
+    logger.debug('Applying IPTables rules [{0}] to {1}'.format(
+        request.param, controller.data['ip']))
+    with controller.ssh() as remote:
+        remote.check_call(tbl_modif)
```

Comment:
Command to execute contains ';' as delimiter. This mean, that only if last command returns not zero exit code - exception will generate. It's better to join commands with `&&` or call each separately, like:
```python
with controller.ssh() as remote:
    for cmd in cmds:
        remote.check_call(cmd)
```